bundles/core/org.eclipse.smarthome.core.thing.test/src/test/java/org/eclipse/smarthome/core/thing/internal/firmware/FirmwareUpdateServiceTest.java

```diff
@@ -118,25 +118,34 @@
     @Before
     public void setup() {
         initMocks(this);
+
         Map<String, String> props1 = new HashMap<>();
         props1.put(Thing.PROPERTY_FIRMWARE_VERSION, V111);
         props1.put(Thing.PROPERTY_MODEL_ID, MODEL1);
+        props1.put(Thing.PROPERTY_VENDOR, VENDOR1);
         thing1 = ThingBuilder.create(THING_TYPE_UID1, THING1_ID).withProperties(props1).build();
+
         Map<String, String> props2 = new HashMap<>();
         props2.put(Thing.PROPERTY_FIRMWARE_VERSION, V112);
-        props2.put(Thing.PROPERTY_MODEL_ID, MODEL2);
+        props2.put(Thing.PROPERTY_MODEL_ID, MODEL1);
+        props2.put(Thing.PROPERTY_VENDOR, VENDOR1);
         thing2 = ThingBuilder.create(THING_TYPE_UID1, THING2_ID).withProperties(props2).build();
+
         Map<String, String> props3 = new HashMap<>();
         props3.put(Thing.PROPERTY_FIRMWARE_VERSION, VALPHA);
+        props3.put(Thing.PROPERTY_MODEL_ID, MODEL2);
+        props3.put(Thing.PROPERTY_VENDOR, VENDOR2);
         thing3 = ThingBuilder.create(THING_TYPE_UID2, THING3_ID).withProperties(props3).build();
 
-        firmwareUpdateService = new FirmwareUpdateService();
+        firmwareUpdateService = new FirmwareUpdateServiceImpl();
 
         safeCaller = getService(SafeCaller.class);
         assertNotNull(safeCaller);
 
         firmwareUpdateService.setSafeCaller(safeCaller);
 
+        firmwareUpdateService.setFirmwareRegistry(mockFirmwareRegistry);
```

Comment:
since it is an osgi test why dont we register the mock as a service?

Reply:
I found this test with part of the services mocked and set manually and the others were injected. As the most important services were mocked I decided to continue the test like this and mock as much as I can. 
I agree that the better approach would be to register the mocks as services, but this was not possible, because of the large number of mostly static dependencies, which would have introduced more sophisticated operations on the Firmware update service component.

At the end, now I am mocking all the dependencies and setting them manually (instead of the SafeCaller that can be mocked as well, but will generate API restriction warning and since we are still JavaOsgiTest it does not make sense), but there is a `FrameworkUtil.getBundle(clazz)` call in the `ProgressCallbackImpl`, which stops from doing JavaTest instead of JavaOsgiTest.

I would be happy to discuss with @tomhoefer and @sjka for a posible better approach.

P.S. I did some small adjustments on the test, so please let me update the PR before starting the discussion.